### PR TITLE
Working branch

### PR DESCRIPTION
**Pull request by MariamHasanat**
This branch has the main requirements of the task _My_Note_Keeper_ for **node.js** exercise :
 
image

the **requirements** : 
image

### CODE REVIEW

File: model/note-model.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    content: {
+        type: String,
+        require: true
+    },
+    creationDate: {
+        type: Date, 
+    }
+});
+
+const Note = mongoose.model('Note', noteSchema);// this will create a collection in DataBase and return it with a set of methods to deal with this collection
```
Comment (m5atib) [inline model/note-model.js]: Could you please remove this comment and all of comments in your project?

File: controller/control-collection.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+const Note = require('../model/note-model');
+
+const createNote = (note) => {
+    const newNote = new Note({
+        title: note.title,
+        content: note.content,
+        creationDate: Date.now()
+    })
+    console.log(newNote);
```
Comment (m5atib) [inline controller/control-collection.js]: Please remove any log from you code

File: controller/control-collection.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    return newNote.save()
+        .then(
+            () => { return true }
+        )
```
Comment (m5atib) [inline controller/control-collection.js]: For better UX it's better to return the created object so the user does not have to hit get all notes request again
`return await newNote.save()`

File: controller/control-collection.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+}
```
Comment (m5atib) [inline controller/control-collection.js:14]: Can you please go with try catch to handle error?

File: controller/control-collection.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+const readNotes = async () => {
```
Comment (m5atib) [inline controller/control-collection.js]: Prefer to rename this function to `getNotes`

File: controller/control-collection.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    const notes = await Note.find();
+    if (notes) {
+        return notes;
+    } else {
+        return null
+    }
+}
```
Comment (m5atib) [inline controller/control-collection.js]: Can you please go with try catch to handle errors ?

File: controller/control-collection.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        const updated = await Note.findByIdAndUpdate(id, updatedNote);
+        if (!updated) {
+            console.log('failed to update');
+            return null
+        } else {
+            return updated
+        }
+
+    } catch (error) {
+        console.log('failed to update : \n', error);
```
Comment (m5atib) [inline controller/control-collection.js]: Can you please throw and return an error response here?

File: controller/control-collection.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+const updateNote = async (id, newData) => {
+    try {
+        const updatedNote = {
+            title: newData.title,
+            content: newData.content,
+            creationDate: newData.creationDate
+        }
+        const updated = await Note.findByIdAndUpdate(id, updatedNote);
+        if (!updated) {
+            console.log('failed to update');
```
Comment (m5atib) [inline controller/control-collection.js]: Same this here (return an error)

File: controller/control-collection.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    } catch (error) {
+        console.log('failed to update : \n', error);
+    }
+}
+
+const deleteNote = async (id) => {
+    try {
+        await Note.findByIdAndDelete(id);
+    } catch (error) {
+        console.log('error in delete ', error);
```
Comment (m5atib) [inline controller/control-collection.js]: Same here

File: index.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+const cors = require('cors')
+require('dotenv').config();
+const  router  = require('./routes/note-router');
+
+
+const app = express();
+app.use(express.json());
+app.use(cors());
+
+// console.log(router);
```
Comment (m5atib) [inline index.js]: Could you please remove any commented line

File: index.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+app.use('/notes', router);
+
+const port = 3001;
+
+app.get('/', (req, res) => {
+    console.log('connected to server successfully');
+})
+const dbConnect = () => {
+    mongoose.connect('mongodb://127.0.0.1:27017/My_Note_Keeper')
```
Comment (m5atib) [inline index.js]: Could you please move this into .env file

**Comment by MariamHasanat** [comment]
@m5atib  all change requests are done successfully.

**Review by m5atib — APPROVED** [review]

**Pull request merged**